### PR TITLE
Launcher recovery on beater panic

### PR DESCRIPTION
**Pull request by amirbenun**

### CODE REVIEW

**Review by yashtewari — APPROVED** [review]
Nice

File: launcher/launcher.go [hunk 1]
```diff
@@ -132,6 +132,11 @@ func (l *launcher) runBeater() error {
 	l.wg.Add(1)
 	go func(beater beat.Beater) {
 		l.log.Info("Launcher is running the new created Beater")
+		defer func() {
```
Comment (uri-weisman) [inline launcher/launcher.go:135]: Very nice!
Do we count on the agent to re-run cloudbeat or do we want to apply a similar logic of our own?

Comment (yashtewari) [reply launcher/launcher.go:135]: At the moment `type launcher` functionality is reserved for re-instantiating `type cloudbeat` with configuration updates.

It makes more sense to propagate the error up to the Agent and let the Agent restart the cloudbeat binary as a whole. This lets the Agent do its own handling on the error (at least conceptually), and it's a more complete restart.

Comment (uri-weisman) [reply launcher/launcher.go:135]: That sounds right.
Is it actually happening? propagating an error to the agent results in a cloudbeat restart?

Comment (yashtewari) [reply launcher/launcher.go:135]: Yes, you can observe a cloudbeat restart loop on the Agent if you return an error from the `Run` method.

**Pull request merged**